### ISSUE TITLE
Trident has a chance of disappearing when thrown

### ISSUE THREAD

**Issue by NjlsShade** [GeyserMC/Geyser#1963]
**Describe the bug**

When releasing a trident to throw, it will sometimes dissapear from the hotbar. It then can't be used until moving another item to occupy its slot on the hotbar.

**To Reproduce**

1. Load into any creative game (tested on a remote server and local, network latency does not seem to be a factor here)
2. Grab a trident and make multiple attempts to trow it

**Expected behavior**

Trident should not disappear

**Screenshots / Videos**

https://user-images.githubusercontent.com/15788214/108647093-6a6a0880-747d-11eb-843c-c21399097cc8.mp4

**Server Version and Plugins**

- Running the latest Java server 1.16.5
- Issue occurs on both the latest master build (Feb 21 18:57) and the latest server-inventory build (Feb 20
23:54)

- Geyser version 1.2.0-SNAPSHOT (git-master-f480fbf)
- Java: 1.16.4 - 1.16.5

**Minecraft: Bedrock Edition Version**

Running on Linux via the mcpelauncher build here: https://github.com/ChristopherHX/linux-packaging-scripts/releases
- Bedrock: 1.16.100/1.16.101 - 1.16.200/1.16.201

**Additional Context**

Having the loyalty enchantment is not required to reproduce, just used to make the demonstration in the video eaiser

**Comment by PrestonJude** [comment]
I would probably say that's a Minecraft error

**Comment by NjlsShade** [comment]
Just to add, I've noticed this issue with placed blocks as well. If you place a block right before an entity moves to the location you were targeting the Bedrock client removes the item from the hotbar, however, the block does not place. I'd think desync with the Java server as the entity was in the way Java side.

Just as with the trident, if you occupy the hotbar slot the block will come back. I suspect the issue happens with the trident due to the Bedrock client successfully throwing while your Java instance did not have the trident charged enough, or maybe did not get the call to use the trident at all?

Ultimately not sure if what I've expressed is accurate or even something in the scope of Geyser vs just a Bedrock is different than Java issue but just wanted to add new info I observed.
